parlai/core/torch_generator_agent.py

```diff
@@ -979,12 +1019,13 @@ def _generate(self, batch, beam_size, max_ts):
         if batch.text_vec is not None:
             dev = batch.text_vec.device
         else:
+            assert batch.label_vec is not None, "need label_vec for _generate"
```

Comment:
when is batch.text_vec None but batch.label_vec exists?

Reply:
if we're in an image-based task, e.g. coco captions